spec/features/course/leaderboard_viewing_spec.rb

```diff
@@ -10,9 +10,43 @@
     end
 
     context 'As a student' do
-      let(:user) { create(:course_student, :approved, course: course).user }
-      scenario 'I can view the leaderboard' do
+      let(:students) { create_list(:course_student, 2, :approved, course: course) }
+      let(:user) { students[0].user }
+      before do
+        create(:course_user_achievement, course_user: students[0])
```

Comment:
does this necessarily have to be a before block? I think each scenario should be self-contained where possible.